plugins/zstack.py

```diff
@@ -0,0 +1,372 @@
+The software is provided "as is", without warranty of any kind,
+express or implied, including but not limited to the warranties of
+merchantability, fitness for a particular purpose and non-infringement.
+In no event shall the authors be liable for any claim, damages or
+other liability, whether in an action of contract, tort or otherwise,
+arising from, out of or in connection with the software or the use or
+other dealings in the software.
+'''
+
+from __future__ import division
+
+from collections import OrderedDict
+import logging
+import math
+import copy
+import numpy as np
+from odemis import model, dataio, acq
+from odemis.acq import stream
+from odemis.util import driver
+from odemis.acq.stream import MonochromatorSettingsStream, ARStream, \
+    SpectrumStream, UNDEFINED_ROI, StaticStream
+import odemis.gui
+from odemis.gui.conf import get_acqui_conf
+from odemis.gui.plugin import Plugin, AcquisitionDialog
+from odemis.util.dataio import splitext
+import os
+import time
+import wx
+from odemis.model._dataflow import DataArray
+
+
+class ZStackPlugin(Plugin):
+    name = "Z Stack"
+    __version__ = "1.0"
+    __author__ = u"Anders Muskens"
+    __license__ = "Public domain"
+
+    # Describe how the values should be displayed
+    # See odemis.gui.conf.data for all the possibilities
+    vaconf = OrderedDict((
+        ("numberOfAcquisitions", {
+            "control_type": odemis.gui.CONTROL_INT,  # no slider
+        }),
+        ("filename", {
+            "control_type": odemis.gui.CONTROL_SAVE_FILE,
+        }),
+        ("expectedDuration", {
+        }),
+    ))
+
+    def __init__(self, microscope, main_app):
+        super(ZStackPlugin, self).__init__(microscope, main_app)
+        # Can only be used with a microscope
+        main_data = self.main_app.main_data
+        
+        if not microscope or main_data.focus is None:
+            return
+
+        self.focus = main_data.focus
+        self._zrange = self.focus.axes['z'].range
+        zunit = self.focus.axes['z'].unit
+        self.old_pos = self.focus.position.value
+        self.zstart = model.FloatContinuous(self.old_pos['z'], range=self._zrange, unit=zunit)
+        self.zstep = model.FloatContinuous(1e-6, range=(-1e-5, 1e-5), unit=zunit, setter=self._setZStep)
+        self.numberofAcquisitions = model.IntContinuous(3, (2, 999), setter=self._setNumberOfAcquisitions)
+
+        self.filename = model.StringVA("a.h5")
+        self.expectedDuration = model.VigilantAttribute(1, unit="s", readonly=True)
+
+        self.zstep.subscribe(self._update_exp_dur)
+        self.numberofAcquisitions.subscribe(self._update_exp_dur)
+        
+        # Metadata for the acquisition
+        self._metadata = None
+        
+        self._acq_streams = None
+
+        self._dlg = None
+        self.addMenu("Acquisition/ZStack...\tCtrl+T", self.start)
+        
+    def _acqRangeIsValid(self, acq_range):
+        return acq_range <= self._zrange[1] and acq_range >= self._zrange[0]
```

Comment:
You can write a little more clearly:
`self._zrange[0] <= acq_range <= self._zrange[1]`